packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -45,18 +47,23 @@ function AnimatedNumber({ duration = 1000, value }) {
   }
 
   useEffect(() => {
+    // If we already animated the number once, don't observe intersection
+    // This could happen if the value prop updates, but page did not refresh
+    if (animated) return
+
     const numElement = numRef.current
 
     const intersectionObserver = new window.IntersectionObserver(entries => {
       // If intersectionRatio is 0, the target is out of view and we do not need to do anything.
       if (entries[0].intersectionRatio <= 0) return
 
-      // Once target element is in viewport, animate it only once
-      if (!prefersReducedMotion) {
+      // Once target element is in viewport, animate it then unobserve
+      if (!prefersReducedMotion && !animated) {
         animateValue()
         intersectionObserver.unobserve(numElement)
       } else {
         lessAnimation()
+        intersectionObserver.unobserve(numElement)
```

Comment:
[minor] I assume that, logic-wise, this means that `intersectionObserver.unobserve(numElement)` could be placed outside the `if ()` block, since it always triggers no matter the if() condition.